worldwind-gt-geopkg/src/main/java/gov/nasa/worldwind/geopkg/mosaic/GeoPackageReader.java

```diff
@@ -90,21 +90,44 @@
     protected File sourceFile;
 
     protected Map<String, TileEntry> tiles = new HashMap<String, TileEntry>();
+    
+    // TODO: store these gridRanges in the TileEntry class
+    protected Map<String, GridEnvelope2D> gridRanges = new HashMap<String, GridEnvelope2D>();
 
     public GeoPackageReader(Object source, Hints hints) throws IOException {
         coverageFactory = CoverageFactoryFinder.getGridCoverageFactory(this.hints);
         sourceFile = GeoPackageFormat.getFileFromSource(source);
         GeoPackage file = new GeoPackage(sourceFile);
 
         try {
-            // Set the coverage name to the name of the the first raster tileset.
             coverageName = null;   // default value was "geotools_coverage"
-            for (TileEntry tile : file.tiles()) {
-                tiles.put(tile.getTableName(), tile);
+            for (TileEntry tileset : file.tiles()) {
+                // Map the tileset to the coverage name (table name)
+                tiles.put(tileset.getTableName(), tileset);
+
+                // Set the default coverage name to the name of the the first raster tileset.
                 if (coverageName == null) {
-                    coverageName = tile.getTableName();
+                    coverageName = tileset.getTableName();
                 }
+
+                // TODO: compute the gridRange in the GeoPackage class
+                List<TileMatrix> matricies = tileset.getTileMatricies();
+                TileMatrix matrix = matricies.get(matricies.size() - 1);
+                int maxZoomLevel = matrix.getZoomLevel();
+                int minCol = file.getTileBound(tileset, maxZoomLevel, false, false);   // booleans: isMax, isRow
+                int maxCol = file.getTileBound(tileset, maxZoomLevel, true, false);
+                int minRow = file.getTileBound(tileset, maxZoomLevel, false, true);  
+                int maxRow = file.getTileBound(tileset, maxZoomLevel, true, true);  
+                int numCols = (maxCol - minCol) + 1;
+                int numRows = (maxRow - minRow) + 1;
+
+                GridEnvelope2D gridRange = new GridEnvelope2D(new Rectangle(
+                        numCols * matrix.getTileWidth(),
+                        numRows * matrix.getTileHeight()));
+                
```

Comment:
I'm been looking into it since your original comment. So far nearly all instances have been on the span, but its a large domain to inspect. I cannot identify an alternative approach to manifesting this data, nor can I propose candidate values for a top/left in this context other than 0,0.